### PR TITLE
[DROOLS-1356] add command to advance pseudo-clock

### PR DESCRIPTION
**Pull request by mariofusco**

### CODE REVIEW

**Comment by mariofusco** [comment]
ok to test

File: drools-core/src/main/java/org/drools/core/command/runtime/AdvanceSessionTimeCommand.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    public Long execute(Context context ) {
+        KieSession ksession = ((RegistryContext)context).lookup( KieSession.class );
+        SessionPseudoClock sessionClock = ksession.<SessionPseudoClock>getSessionClock();
+
+        sessionClock.advanceTime( amount, unit );
+        long result = sessionClock.getCurrentTime();
+
+        ExecutionResultImpl results = ((RegistryContext)context).lookup( ExecutionResultImpl.class );
+        if ( results != null ) {
+            results.getResults().put( this.outIdentifier, result );
```
Comment (tarilabs) [inline drools-core/src/main/java/org/drools/core/command/runtime/AdvanceSessionTimeCommand.java:72]: Sorry to insist :) but with reference to draft from here:
https://github.com/tarilabs/drools/blob/patch-pseudoclockcmds/drools-core/src/main/java/org/drools/core/command/runtime/PseudoClockAdvanceTimeCommand.java#L58-L63

I think the out identifier for this command should be optional, and if is valorized (meaning the user is interested in the out value) then we place the result in the returning results.

This is a bit different from the `GetSessionTimeCommand` where we can always expect to have the out identifier.

What do you think, please?

**Review by tarilabs — CHANGES_REQUESTED** [review]
I think the public factory should update for the option to specify the out identifier?

File: drools-core/src/main/java/org/drools/core/command/impl/CommandFactoryServiceImpl.java [hunk 2]
```diff
@@ -357,7 +357,7 @@ public Command newEnableAuditLog( String filename ) {
     }
 
     @Override
-    public Command<Long> newAdvanceSessionTime(String outIdentifier, long amount, TimeUnit unit) {
+    public Command<Long> newAdvanceSessionTime(long amount, TimeUnit unit, String outIdentifier) {
```
Comment (tarilabs) [inline drools-core/src/main/java/org/drools/core/command/impl/CommandFactoryServiceImpl.java]: thanks :)

**Review by tarilabs — APPROVED** [review]
+1

**Pull request merged**